loader/src/sources/cdc/api.js

```diff
@@ -0,0 +1,282 @@
+        latitude: base.latitude,
+      },
+      info_phone: base.loc_phone,
+      info_url: base.web_address,
+      meta,
+
+      availability: {
+        source: "cdc",
+        checked_at: new Date().toISOString(),
+        valid_at: formatValidAt(productList),
+        available_count: getAvailableCount(productList),
+        available: formatAvailable(productList),
+        capacity: formatCapacity(productList),
+        products: formatProductTypes(productList),
+      },
+    };
+  });
+  return result;
+}
+
+const systemNameRe = {
+  costco: /^Costco/i,
+  cvs: /^CVS/i,
+  kroger: /^Kroger/i,
+  publix: /^Publix/i,
+  rite_aid: /^Rite Aid/i,
+  safeway: /^SAFEWAY/i,
+  sams_club: /^Sams Club/i,
+  walgreens: /^Walgreens/i,
+  walmart: /^Walmart/i,
+};
+
+function getStoreExternalId(store) {
+  const m = store.loc_store_no.match(/^[A-Z]*(\d+)$/); // handle cases like RA105587 -> 105587
+  if (!m) {
+    return null;
+  }
+  const storeNumber = m[1];
+
+  for (const system in systemNameRe) {
+    if (store.loc_name.match(systemNameRe[system])) {
+      return [system, storeNumber];
+    }
+  }
+}
+
+function formatValidAt(products) {
+  const dates = products.map((p) => p.quantity_last_updated);
+  dates.sort();
+  return dates[dates.length - 1];
+}
+
+function getAvailableCount(products) {
+  if (!Array.isArray(products)) {
+    products = [products];
+  }
+
+  let sum = 0;
+  for (const product of products) {
+    const supplyLevel = parseInt(product.supply_level, 10);
+    if (supplyLevel > 0) {
+      sum += supplyLevel;
+    } else {
+      sum += product.in_stock ? 1 : 0;
+    }
+  }
+  return sum;
+}
+
+function formatAvailable(products) {
+  return getAvailableCount(products) > 0 ? Available.yes : Available.no;
+}
+
+const ndcLookup = {
+  // from https://www.cdc.gov/vaccines/programs/iis/COVID-19-related-codes.html
+  // and https://www2a.cdc.gov/vaccines/iis/iisstandards/vaccines.asp?rpt=ndc
+  [normalizeNDC("00310-1222-10")]: "astra_zeneca", //use
+  [normalizeNDC("00310-1222-15")]: "astra_zeneca", //sale
+  [normalizeNDC("59267-1000-01")]: "pfizer", //use
+  [normalizeNDC("59267-1000-02")]: "pfizer", //sale
+  [normalizeNDC("59267-1000-03")]: "pfizer", //sale
+  [normalizeNDC("59676-0580-05")]: "jj", //use
+  [normalizeNDC("59676-0580-15")]: "jj", //sale
+  [normalizeNDC("80631-0100-01")]: "novavax", //sale
+  [normalizeNDC("80631-0100-10")]: "novavax", //use
+  [normalizeNDC("80777-0273-10")]: "moderna", // use
+  [normalizeNDC("80777-0273-15")]: "moderna", // use
+  [normalizeNDC("80777-0273-98")]: "moderna", //sale
+  [normalizeNDC("80777-0273-99")]: "moderna", //sale
+};
+function normalizeNDC(ndcCode) {
+  // Note: this normalization makes consistent string values, not NDCs
+  const parsed = ndcCode.match(/^(\d+)-(\d+)-(\d+)$/);
+  if (!parsed) {
+    throw new Error(`Unexpected NDC format '${ndcCode}'`);
+  }
+  return parsed
+    .slice(1)
+    .map((n) => parseInt(n, 10).toString())
+    .join("~");
+}
```

Comment:
Not a big deal, but it *might* be worthwhile to move this NDC stuff up to `utils` or its own module or something.

Reply:
I have some hesitation here just because I think this code is a bit hacky and not so robust to arbitrary NDC formats.

Reply:
That’s fair, and a good reason to keep it here for now.